core/src/main/scala/org/apache/spark/deploy/DeployMessage.scala

```diff
@@ -158,6 +158,8 @@ private[deploy] object DeployMessages {
 
   case class ApplicationRemoved(message: String)
 
+  case class WorkerRemoved(id: String, host: String, message: String)
```

Comment:
There are some overlap but not quite straight forward, I can remove some of the SlaveLost(...workerLost) messages as a followup work.
But for this PR, I'd like to keep the scope to only add a new message type and don't change the existing messages, to make the review easier.